aws/resource_aws_batch_compute_environment.go

```diff
@@ -0,0 +1,429 @@
+		}
+		if v, ok := computeResource["image_id"]; ok {
+			input.ComputeResources.ImageId = aws.String(v.(string))
+		}
+		if v, ok := computeResource["spot_iam_fleet_role"]; ok {
+			input.ComputeResources.SpotIamFleetRole = aws.String(v.(string))
+		}
+		if v, ok := computeResource["tags"]; ok {
+			input.ComputeResources.Tags = tagsFromMapGeneric(v.(map[string]interface{}))
+		}
+	}
+
+	log.Printf("[DEBUG] Create compute environment %s.\n", input)
+
+	if _, err := conn.CreateComputeEnvironment(input); err != nil {
+		return err
+	}
+
+	d.SetId(computeEnvironmentName)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{batch.CEStatusCreating},
+		Target:     []string{batch.CEStatusValid},
+		Refresh:    resourceAwsBatchComputeEnvironmentStatusRefreshFunc(d, meta),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 5 * time.Second,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return err
+	}
+
+	return resourceAwsBatchComputeEnvironmentRead(d, meta)
+}
+
+func resourceAwsBatchComputeEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).batchconn
+
+	computeEnvironmentName := d.Get("compute_environment_name").(string)
+
+	input := &batch.DescribeComputeEnvironmentsInput{
+		ComputeEnvironments: []*string{
+			aws.String(computeEnvironmentName),
+		},
+	}
+
+	log.Printf("[DEBUG] Read compute environment %s.\n", input)
+
+	result, err := conn.DescribeComputeEnvironments(input)
+	if err != nil {
+		return err
+	}
+
+	if len(result.ComputeEnvironments) == 0 {
+		return fmt.Errorf("One compute environment is expected, but AWS return no compute environment")
+	}
+	computeEnvironment := result.ComputeEnvironments[0]
+
+	d.Set("service_role", computeEnvironment.ServiceRole)
+	d.Set("state", computeEnvironment.State)
+	d.Set("type", computeEnvironment.Type)
+
+	if *(computeEnvironment.Type) == "MANAGED" {
+		computeResource := computeEnvironment.ComputeResources
+
+		d.Set("compute_resources", []interface{}{
+			map[string]interface{}{
+				"bid_percentage":      computeResource.BidPercentage,
+				"desired_vcpus":       computeResource.DesiredvCpus,
+				"ec2_key_pair":        computeResource.Ec2KeyPair,
+				"image_id":            computeResource.ImageId,
+				"instance_role":       computeResource.InstanceRole,
+				"instance_type":       schema.NewSet(schema.HashString, flattenStringList(computeResource.InstanceTypes)),
+				"max_vcpus":           computeResource.MaxvCpus,
+				"min_vcpus":           computeResource.MinvCpus,
+				"security_group_ids":  schema.NewSet(schema.HashString, flattenStringList(computeResource.SecurityGroupIds)),
+				"spot_iam_fleet_role": computeResource.SpotIamFleetRole,
+				"subnets":             schema.NewSet(schema.HashString, flattenStringList(computeResource.Subnets)),
+				"tags":                tagsToMapGeneric(computeResource.Tags),
+				"type":                computeResource.Type,
+			},
+		})
+	}
```

Comment:
Last thing to do would be to move this to a flatten function I think :)
It would then allow us to manage optional fields more easily, and be consistent with other parts of the code.

Once done, 'will run the tests and merge if it's all ok :)